wurst/objects/items/PlagueBomb.wurst

```diff
@@ -0,0 +1,163 @@
+
+
+@compiletime function createPlagueBombAbility()
+    new PlagueBomb(PLAGUE_ABIL)
+
+@compiletime function createPlagueBombItem()
+    new ItemDefinition(ITEM_PLAGUE_BOMB, ItemIds.PlagueBomb)
+    ..setName(TOOLTIP)
+    ..setAbilities(ITEM_ABILITIES)
+    ..setTooltipExtended(TOOLTIP_EXTENDED)
+    ..setModelUsed(MODEL)
+    ..setTintingColor1Red(0)
+    ..setTintingColor2Green(70)
+    ..setTintingColor3Blue(0)
+    ..setGoldCost(0)
+    ..setInterfaceIcon(ICON)
+    ..setDescription(DESCRIPTION)
+    ..setDroppedWhenCarrierDies(true)
+    ..setScalingValue(0.80)
+    ..setActivelyUsed(true)
+    ..setCooldownGroup("charged")
+    ..setHitPoints(75)
+    ..setLumberCost(68)
+    ..setLevelUnclassified(2)
+    ..setPerishable(false)
+    ..setPriority(41)
+    ..setTooltipBasic("Trade for "+TOOLTIP)
+    ..setNumberofCharges(1)
+    ..setPerishable(true)
+    ..setCanBeDropped(true)
+    ..setIgnoreCooldown(true)
+
+//Create the projectile which drives the rest of the spell
+function createMissile(unit caster, vec3 targetPos)
+    vec3 initialPos = caster.getPos().withTerrainZ(MISSILE_FLY_HEIGHT)
+    real distance = targetPos.distanceTo2d(initialPos)
+    angle currentAngle = caster.getFacingAngle()
+
+    effect missileEffect = addEffect(MODEL , initialPos)
+    vec3 distanceVector = targetPos - initialPos
+
+    //Move the projectile to its target location.
+    doPeriodically(0.02) (CallbackPeriodic cb) ->
+        vec3 currentPos = missileEffect.getPos3()
+        currentPos += PROJECTILE_INTERVAL * distanceVector
+        //Add rotation to the projectile
+        currentAngle += 5..asAngleDegrees()
+
+        missileEffect.setOrientation(currentAngle, currentAngle, currentAngle)
+        missileEffect.setPos(currentPos)
+
+        //If the projectile is at target position.
+        if currentPos.distanceTo(targetPos) < 0.1
+            //Stop the loop.
+            destroy cb
+            //Destroy the effect.
+            missileEffect.destr()
+            //Create the plague cloud at target location.
+            createPlagueCloud(targetPos)
+            //Start the process of infecting meat/buildings.
+            replaceMeat(targetPos.toVec2(), caster)
+        
+
+//Creates and destroys the plague cloud effect at target location.
+function createPlagueCloud(vec3 targetPos)
+    effect plagueEffect = addEffect(EFFECT, targetPos)
+    plagueEffect.setScale(1)
+    doAfter(EFFECT_DURATION) -> 
+        plagueEffect.destr()
+
+//Finds and replaces all meat on the ground and in buildings.
+function replaceMeat(vec2 targetPos, unit caster)
+    //Filter through all items in the map to get the ones in the ability radius.
+    forItemsInMap() (item i) ->
+        if i.getPos().distanceTo(targetPos) < ABILITY_RADIUS and i.getTypeId() == ITEM_COOKED_MEAT
+            //Remove all cooked meat found on ground and replace with diseased meat.
+            createItem(ITEM_DISEASED_MEAT, i.getPos())
+            i.remove()
```

Comment:
What about the stacked meat? This will replace a 10 stacked cooked meat lying on the ground by a 1 disease meat.